src/SonarQube.TeamBuild.PreProcessor/AnalysisConfigGenerator.cs

```diff
@@ -67,6 +67,13 @@ public static class AnalysisConfigGenerator
                 SonarQubeHostUrl = localSettings.SonarQubeUrl
             };
 
+            // Store whether or not a credential was passed in the command line. This is used to fail-fast when it is provided
+            // in the begin step but not the end step.
+            if (localSettings.CmdLineProperties.TryGetProperty(SonarProperties.SonarUserName, out var _))
```

Comment:
I don't care about the property hence the `out var _`. I don't know if you are aware of this pattern for unused variables.